packages/hardhat-core/src/internal/hardhat-network/provider/provider.ts

```diff
@@ -506,19 +505,14 @@ export async function createHardhatNetworkProvider(
 
 async function makeTracingConfig(
   artifacts: Artifacts | undefined
-): Promise<TracingConfig | undefined> {
+): Promise<TracingConfigWithBuffers | undefined> {
   if (artifacts !== undefined) {
-    const buildInfos = [];
-
     const buildInfoFiles = await artifacts.getBuildInfoPaths();
 
     try {
-      for (const buildInfoFile of buildInfoFiles) {
-        const buildInfo = await fsExtra.readJson(buildInfoFile);
-        if (semver.gte(buildInfo.solcVersion, FIRST_SOLC_VERSION_SUPPORTED)) {
```

Comment:
This check is now done in EDR: https://github.com/NomicFoundation/edr/blob/403bd351ded6a75adddb3879003245e4559e5354/crates/edr_solidity/src/artifacts.rs#L83